### ISSUE TITLE
Instanceof fails due to regex

### ISSUE THREAD

**Issue by qdequippe** [E-vence/SoftDeleteableListenerExtensionBundle#4]
Hi,

I tried to use this bundle but it doesn't work in my case.
Please see below an example with a configuration
```
/**
 * Treasury
 *
 * @ORM\Table()
 * @ORM\Entity(repositoryClass="AppBundle\Repository\TreasuryRepository")
 * @Gedmo\SoftDeleteable(fieldName="deleted", timeAware=false)
 * @Serializer\ExclusionPolicy("all")
 * @ORM\EntityListeners({"AppBundle\EntityListener\TreasuryListener"})
 */
class Treasury
{

...    

    /**
     * @var Adherent
     *
     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Adherent", inversedBy="treasuries", cascade={"persist"})
     * @ORM\JoinColumn(name="adherent_id", referencedColumnName="id", onDelete="SET NULL")
     * @Serializer\Expose()
     * @Serializer\Groups({"list","details"})
     * @Evence\onSoftDelete(type="SET NULL")
     */
    private $adherent;
```
When I try to delete an `Adherent` the `adherent_id` field is not set to null in `Treasury`.
After a few debugging I found this
```
$ns = $manyToOne->targetEntity;
if (!preg_match('#^\\\#', $ns)) {
    $ns = $entityReflection->getNamespaceName().'\\'.$ns;
}

// here $entity = "Adherent" but $ns = "AppBundle\Entity\AppBundle\Entity\Adherent" instead of just 
// "AppBundle\Entity\Adherent", 

if ($entity instanceof $ns) { // so this check is never true
   $objects = $em->getRepository($namespace)->findBy(array(
        $property->name => $entity,
    ));
}
```
https://github.com/E-vence/SoftDeleteableListenerExtensionBundle/blob/master/EventListener/SoftDeleteListener.php#L53

Is there a reason why this regex is used?

**Comment by RubenHarms** [comment]
The purpose of the regex is to detect whether a relative or a absolute  namespace path has been used, however it seems not work in every case. You can bypass the problem, by changing your targetEntity to:
```
* @ORM\ManyToOne(targetEntity="\AppBundle\Entity\Adherent", inversedBy="treasuries", cascade={"persist"})
```
or

```
* @ORM\ManyToOne(targetEntity="Adherent", inversedBy="treasuries", cascade={"persist"})
```

But that's not the real solution for this problem. In the meantime i'll write a replacement for that code.

**Comment by RubenHarms** [comment]
@qdequippe Can you replace the following code:

```
$ns = $manyToOne->targetEntity;
if (!preg_match('#^\\\#', $ns)) {
    $ns = $entityReflection->getNamespaceName().'\\'.$ns;
}
```

To:

```
$ns = null;
$nsOrginal = $manyToOne->targetEntity;
$nsFromRelativeToAbsolute = $entityReflection->getNamespaceName().'\\'.$manyToOne->targetEntity;
$nsFromRoot = '\\'.$manyToOne->targetEntity;
if(class_exists($nsOrginal)){
   $ns = $nsOriginal;
}
elseif(class_exists($nsFromRoot)){
  $ns = $nsFromRoot;
}
elseif(class_exists($nsFromRelativeToAbsolute)){
   $ns = $nsFromRelativeToAbsolute;
}

```

Could you please let me know if this is the solution for the problem?

**Comment by qdequippe** [comment]
@RubenHarms yes it works!

**Comment by RubenHarms** [comment]
Code has merged and released under version 1.2.1

**Comment by qdequippe** [comment]
@RubenHarms sorry I forgot to tell you about variable name here just a small error on `$nsOriginal` without `i` (above it's $nsOrginal)
